MiniApp/Classes/MiniAppAdmobDisplayer.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import GoogleMobileAds
+
+/// Made to be a singleton
+/// It has to be a single instance to avoid multiple instances of Admob SDK
+class MiniAppAdmobDisplayer: NSObject, MiniAppAdDisplayProtocol {
+	var isAdmobInitialized = false
+	var interstitial: GADInterstitial!
+	var rewarded: GADRewardedAd?
+	var onInterstitialClosed: (() -> Void)?
+	var onRewardedClosed: ((MiniAppReward?) -> Void)?
+	var lastReward: MiniAppReward?
+
+	override init() {
+		super.init()
+
+		createAndLoadInterstitial()
+		createAndLoadRewarded()
+	}
+
+	private func createAndLoadRewarded() {
+		rewarded = GADRewardedAd(adUnitID: "ca-app-pub-3940256099942544/1712485313")
+		rewarded?.load(GADRequest())
+	}
+
+	private func createAndLoadInterstitial() {
+		interstitial = GADInterstitial(adUnitID: "ca-app-pub-3940256099942544/4411468910")
+		interstitial.delegate = self
+		interstitial.load(GADRequest())
+	}
+
+	func showInterstitial(onClosed: @escaping (() -> Void), onFailed: @escaping ((Error) -> Void)) {
+		if !isAdmobInitialized {
+			GADMobileAds.sharedInstance().start(completionHandler: {
+				[weak self] _ in
+
+				self?.isAdmobInitialized = true
+				self?.showInterstitial(onClosed: onClosed, onFailed: onFailed)
+			})
+		} else if interstitial.isReady {
+			self.onInterstitialClosed = onClosed
+			// PIERRE: here I need some reference to the viewController displaying the miniApp to allow the `present`
```

Comment:
I understand. We usually take to top most presented controller to present another one. Typically:
```swift
        if #available(iOS 13, *) {
            window = UIApplication.shared.windows.filter { $0.isKeyWindow }.first
        } else {
            window = UIApplication.shared.keyWindow
        }
       [...]
        window?.topController()?.present(interstitial, animated: true)
```